src/mpid/ch4/netmod/ofi/ofi_init.c

```diff
@@ -1589,11 +1607,23 @@ static int addr_exchange_all_vnis(void)
         goto fn_exit;
     }
 
+    int *is_node_roots = NULL;
+    if (MPIR_CVAR_CH4_ROOTS_ONLY_PMI) {
+        MPIR_CHKLMEM_MALLOC(is_node_roots, int *, size * sizeof(int),
+                            mpi_errno, "is_node_roots", MPL_MEM_OTHER);
+        for (int r = 0; r < size; r++) {
+            is_node_roots[r] = 0;
+        }
+        for (int i = 0; i < MPIR_Process.num_nodes; i++) {
+            is_node_roots[MPIR_Process.node_root_map[i]] = 1;
+        }
+    }
+
     /* libfabric uses uniform name_len within a single provider */
     int name_len = MPIDI_OFI_global.addrnamelen;
     int my_len = num_vnis * num_nics * name_len;
-    char *all_names = MPL_malloc(size * my_len, MPL_MEM_ADDRESS);
-    MPIR_Assert(all_names);
+    char *all_names;
+    MPIR_CHKLMEM_MALLOC(all_names, char *, size * my_len, mpi_errno, "all_names", MPL_MEM_OTHER);
```

Comment:
nitpick - `MPL_MEM_ADDRESS` which was used before, is a more accurate category

Reply:
I'll update it.